documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,1243 @@
+        retry_jitter_mode=mqtt5.ExponentialBackoffJitterMode.FULL)
+
+mqtt5_client.start()
+
+```
+
+### Offline Operations Queue
+
+In the v1 SDK, if you're having too many in-flight QoS 1 messages, you can encounter the `too many publishes in Progress` error
+on publishing messages. This is caused by the so-called in-flight publish limit. By default, ther v1 SDK supports
+a maximum of 20 in-flight operations.
+
+The v2 SDK does not limit the number of in-flight messages.
+Additionally, the v2 SDK provides a way to configure which kind of packets will be placed into the offline queue
+when the client is in the disconnected state. The following code snippet demonstrates how to enable storing all packets
+except QOS0 publish packets in the offline queue on disconnect:
+
+#### Example of configuring the offline queue in the v2 SDK
+
+```python
+mqtt5_client = mqtt5_client_builder.mtls_from_path(
+        endpoint="<prefix>-ats.iot.<region>.amazonaws.com",
+        cert_filepath="<certificate file path>",
+        pri_key_filepath="<private key file path>",
+ ...
+        offline_queue_behavior=
+            mqtt5.ClientOperationQueueBehaviorType.FAIL_QOS0_PUBLISH_ON_DISCONNECT)
+
+mqtt5_client.start()
+
+```
+
+> [!NOTE]
+> AWS IoT Core [limits the number of allowed operations per second](https://docs.aws.amazon.com/general/latest/gr/iot-core.html#message-broker-limits).
+> The [get_stats](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.Client.get_stats) method returns
+> the current state of an `awscrt.mqtt5.Client` object's queue of operations in
+> [OperationStatisticsData](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.OperationStatisticsData),
+> which may help with tracking the number of in-flight messages.
+
+``` python
+op_stats_data = mqtt5_client.get_stats()
+print(
+    "Client operations queue statistics:\n" +
+    "incomplete_operation_count:" + op_stats_data.incomplete_operation_count() + "\n"
+    "incomplete_operation_size: " + op_stats_data.incomplete_operation_size() + "\n"
+    "unacked_operation_count: " + op_stats_data.unacked_operation_count() + "\n"
+    "unacked_operation_size: " + op_stats_data.unacked_operation_size() + "\n")
+
+```
+
+For more information, see [withOfflineQueueBehavior documentation](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.ClientOptions).
+
+For the list of the supported offline queue behaviors and their descriptions,
+see [ClientOfflineQueueBehavior types documentation](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.ClientOperationQueueBehaviorType).
+
+### Operation Timeouts
+
+In the v1 SDK, all operations (*publish*, *subscribe*, *unsubscribe*) will not timeout unless
+you define a timeout for them. If no timeout is defined, there is a possibility that an operation will wait forever
+for the server to respond and block the calling thread indefinitely.
```

Comment:
This is probably specific for java sdk v1, so I'd check if it's true for python.